metadata/metadata.go

```diff
@@ -192,6 +193,7 @@ type Resource interface {
 	Proto() proto.Message
 	UpdateStatus(pb.ResourceStatus) error
 	UpdateSchedule(string) error
+	Update(ResourceLookup) error
```

Comment:
Update should take a ResourceLookup and a Resource, we need the resource to be updated with.

Reply:
Is this likely due to a misunderstanding on my end, but my thought here was:
* `res` is the object in memory from the request (e.g. `res Resource` in `genericCreate`)
* if we now a record exists in etcd with the key `res.ID()`, then calling `res.Update(serv.lookup)` allows the `Update` method to fetch that existing record and merge the two according to whatever rules we have for the specific resource
* after the call to `Update` completes without err, we know `res` now represents the updated value for the resource, which is set back to etcd

Reply:
@simba-git, never mind: I chatted with @sdreyer about this and I now see the assumption I was making about the `Update` method (i.e. that `res` would always represent the inbound request message and not also (possibly) the existing record in etcd). I'll update the PR to reflect this new understanding.